packages/manager/apps/web-hosting/src/data/hooks/webHostingAttachedDomain/useWebHostingAttachedDomain.ts

```diff
@@ -44,23 +48,25 @@ export const useWebHostingAttachedDomain = (
       ),
   });
 
-  const fetchAllPages = useCallback(() => {
-    if (!allPages) {
-      setAllPages(true);
-    }
-  }, [allPages, setAllPages]);
-  useEffect(() => {
-    if (allPages && query.hasNextPage && !query.isFetchingNextPage) {
-      query.fetchNextPage();
-    }
-  }, [query.data]);
-  useEffect(() => {
-    if (!shouldFetchAll) {
-      setAllPages(false);
-    }
-  }, []);
+  const allDataQuery = useQuery<ApiResponse, Error>({
+    queryKey: [...getWebHostingAttachedDomainQueryKey(true), 'all'],
+    queryFn: async () => {
+      const response = await getWebHostingAttachedDomain({
+        pageSize: 500,
+      });
+      return response as ApiResponse;
+    },
+    enabled: false,
+  });
+
+  const fetchAllPages = useCallback(async () => {
+    return allDataQuery.refetch() as Promise<
```

Comment:
every time this will be called it will refetch all the data, also if you already have all data fetched it won't reuse it